src/routes/open/container/Open.tsx

```diff
@@ -82,17 +83,32 @@ export const createSafe = async (values: CreateSafeValues, userAccount: string):
   const ownerAddresses = getAccountsFrom(values)
   const safeCreationSalt = getSafeCreationSaltFrom(values)
   const deploymentTx = getSafeDeploymentTransaction(ownerAddresses, confirmations, safeCreationSalt)
-
-  const receipt = await deploymentTx
+  let receiptResult
+  deploymentTx
     .send({
       from: userAccount,
       gas: values?.gasLimit,
     })
     .once('transactionHash', (txHash) => {
       saveToStorage(SAFE_PENDING_CREATION_STORAGE_KEY, { txHash, ...values })
+      txMonitor({ sender: userAccount, hash: txHash, data: deploymentTx.encodeABI() }, (txReceipt) => {
+        receiptResult = txReceipt
+      })
+    })
+    .then((txReceipt) => {
+      receiptResult = txReceipt
     })
```

Comment:
Let's move the `new Promise` to the beginning of the function and add `.catch((err) => reject(err))` here.
So that it can be properly caught [here](https://github.com/gnosis/safe-react/blob/dev/src/routes/opening/index.tsx#L201).